components/board/pi/common/common.go

```diff
@@ -17,12 +17,13 @@ var ModelName = resource.NewDefaultModel("pi")
 
 // ServoConfig is the config for a pi servo.
 type ServoConfig struct {
-	Pin       string   `json:"pin"`
-	Min       int      `json:"min,omitempty"`
-	Max       int      `json:"max,omitempty"`
-	StartPos  *float64 `json:"starting_position_degs,omitempty"`
-	HoldPos   *bool    `json:"hold_position,omitempty"` // defaults True. False holds for 500 ms then disables servo
-	BoardName string   `json:"board"`
+	Pin         string   `json:"pin"`
+	Min         int      `json:"min,omitempty"`
+	Max         int      `json:"max,omitempty"` // specifies a user inputted position limitation
+	StartPos    *float64 `json:"starting_position_degs,omitempty"`
+	HoldPos     *bool    `json:"hold_position,omitempty"` // defaults True. False holds for 500 ms then disables servo
+	BoardName   string   `json:"board"`
+	MaxRotation int      `json:"max_rotation_deg,omitempty"` // specifies a hardware position limitation. Defaults to 180
```

Comment:
[minor] Address the case in `Validate` where `MaxRotation` < `Min`, `MaxRotation` < `Max` and have that send an error message to the user.